shapely/geos.py

```diff
@@ -817,7 +817,13 @@ def __init__(self, dll):
         attr.__name__ = func.__name__
         setattr(self, key, attr)
 
-        for pred in (self.GEOSisClosed,):
+        coveredByKey = 'GEOSCoveredBy'
+        func = getattr(self._lgeos, coveredByKey + '_r')
+        attr = partial(func, self.geos_handle)
+        attr.__name__ = func.__name__
+        setattr(self, coveredByKey, attr)
```

Comment:
Do you know is this was needed? It is done like this for GEOSPolygonize_full above, but there is a comment explicitly stating that that was a problem with the GEOS binary. 

(but not fully familiar with this part of the shapely code)

Reply:
That was a special case. Do you find that we need it here @bhavika?

Reply:
@sgillies @jorisvandenbossche I added that because it seemed to me as if the GEOS binary wasn't exposing `GEOSCoveredBy` and `GEOSCoveredBy_r`. At first, I thought those were missing when I printed out all `keys`, but I have now confirmed that's not an issue (see below). 

In LGEOS330, I added two lines [below this](https://github.com/Toblerity/Shapely/blob/master/shapely/geos.py#L810) which were 

```
keys = list(self._lgeos.__dict__.keys())
print("330" , keys)
``` 

which prints all the methods found in GEOS (I am running 3.7.0) -

<details>
330 ['_name', '_FuncPtr', '_handle', 'GEOSversion', 'initGEOS', 'finishGEOS', 'GEOSGeomFromWKT', 'GEOSGeomToWKT', 'GEOS_setWKBOutputDims', 'GEOSGeomFromWKB_buf', 'GEOSGeomToWKB_buf', 'GEOSCoordSeq_create', 'GEOSCoordSeq_clone', 'GEOSGeom_clone', 'GEOSCoordSeq_destroy', 'GEOSCoordSeq_setX', 'GEOSCoordSeq_setY', 'GEOSCoordSeq_setZ', 'GEOSCoordSeq_setOrdinate', 'GEOSCoordSeq_getX', 'GEOSCoordSeq_getY', 'GEOSCoordSeq_getZ', 'GEOSCoordSeq_getSize', 'GEOSCoordSeq_getDimensions', 'GEOSProject', 'GEOSInterpolate', 'GEOSProjectNormalized', 'GEOSInterpolateNormalized', 'GEOSBuffer', 'GEOSBufferWithStyle', 'GEOSOffsetCurve', 'GEOSBufferParams_create', 'GEOSBufferParams_setEndCapStyle', 'GEOSBufferParams_setJoinStyle', 'GEOSBufferParams_setMitreLimit', 'GEOSBufferParams_setQuadrantSegments', 'GEOSBufferParams_setSingleSided', 'GEOSBufferWithParams', 'GEOSGeom_createPoint', 'GEOSGeom_createLinearRing', 'GEOSGeom_createLineString', 'GEOSGeom_createPolygon', 'GEOSGeom_createCollection', 'GEOSGeom_createEmptyCollection', 'GEOSGeom_destroy', 'GEOSEnvelope', 'GEOSIntersection', 'GEOSConvexHull', 'GEOSDifference', 'GEOSSymDifference', 'GEOSBoundary', 'GEOSUnion', 'GEOSUnaryUnion', 'GEOSUnionCascaded', 'GEOSPointOnSurface', 'GEOSGetCentroid', 'GEOSClipByRect', 'GEOSPolygonize', 'GEOSPolygonize_full', 'GEOSDelaunayTriangulation', 'GEOSVoronoiDiagram', 'GEOSLineMerge', 'GEOSSimplify', 'GEOSTopologyPreserveSimplify', 'GEOSDisjoint', 'GEOSTouches', 'GEOSIntersects', 'GEOSCrosses', 'GEOSWithin', 'GEOSContains', 'GEOSOverlaps', 'GEOSCovers', 'GEOSCoveredBy', 'GEOSEquals', 'GEOSEqualsExact', 'GEOSisEmpty', 'GEOSisValid', 'GEOSisValidReason', 'GEOSisSimple', 'GEOSisRing', 'GEOSisClosed', 'GEOSHasZ', 'GEOSRelate', 'GEOSRelatePattern', 'GEOSRelatePatternMatch', 'GEOSPrepare', 'GEOSPreparedGeom_destroy', 'GEOSPreparedDisjoint', 'GEOSPreparedTouches', 'GEOSPreparedIntersects', 'GEOSPreparedCrosses', 'GEOSPreparedWithin', 'GEOSPreparedContains', 'GEOSPreparedContainsProperly', 'GEOSPreparedOverlaps', 'GEOSPreparedCovers', 'GEOSGeomType', 'GEOSGeomTypeId', 'GEOSGetSRID', 'GEOSSetSRID', 'GEOSGetNumGeometries', 'GEOSGetGeometryN', 'GEOSGetNumInteriorRings', 'GEOSGetInteriorRingN', 'GEOSGetExteriorRing', 'GEOSGetNumCoordinates', 'GEOSGeom_getCoordSeq', 'GEOSGeom_getDimensions', 'GEOSArea', 'GEOSLength', 'GEOSDistance', 'GEOSHausdorffDistance', 'GEOSWKTReader_create', 'GEOSWKTReader_destroy', 'GEOSWKTReader_read', 'GEOSWKTWriter_create', 'GEOSWKTWriter_destroy', 'GEOSWKTWriter_write', 'GEOSWKTWriter_setTrim', 'GEOSWKTWriter_setRoundingPrecision', 'GEOSWKTWriter_setOutputDimension', 'GEOSWKTWriter_getOutputDimension', 'GEOSWKTWriter_setOld3D', 'GEOSWKBReader_create', 'GEOSWKBReader_destroy', 'GEOSWKBReader_read', 'GEOSWKBReader_readHEX', 'GEOSWKBWriter_create', 'GEOSWKBWriter_destroy', 'GEOSWKBWriter_write', 'GEOSWKBWriter_writeHEX', 'GEOSWKBWriter_getOutputDimension', 'GEOSWKBWriter_setOutputDimension', 'GEOSWKBWriter_getByteOrder', 'GEOSWKBWriter_setByteOrder', 'GEOSWKBWriter_getIncludeSRID', 'GEOSWKBWriter_setIncludeSRID', 'GEOSFree', 'GEOSSnap', 'GEOSSharedPaths', 'GEOSNearestPoints', 'GEOSQueryCallback', 'GEOSSTRtree_query', 'GEOSSTRtree_create', 'GEOSSTRtree_insert', 'GEOSSTRtree_remove', 'GEOSSTRtree_destroy', 'GEOSDistanceCallback', 'GEOSSTRtree_nearest_generic', 'GEOSMinimumClearance', 'GEOSRelatePatternMatch_r', 'GEOSSnap_r', 'GEOS_setWKBOutputDims_r', 'GEOSGeom_createPoint_r', 'GEOSCoordSeq_getZ_r', 'GEOSProjectNormalized_r', 'GEOSUnionCascaded_r', 'GEOSPolygonize_full_r', 'GEOSPreparedContainsProperly_r', 'GEOSContains_r', 'GEOSGeom_createCollection_r', 'GEOSGeom_clone_r', 'GEOSCoordSeq_clone_r', 'GEOSBufferParams_setMitreLimit_r', 'GEOSGeomType_r', 'GEOSCoordSeq_getY_r', 'GEOSDistance_r', 'GEOSWKBWriter_getIncludeSRID_r', 'GEOSGetNumGeometries_r', 'GEOSConvexHull_r', 'GEOSPreparedCovers_r', 'GEOSWKTWriter_destroy_r', 'GEOSIntersects_r', 'GEOSPreparedContains_r', 'GEOSisEmpty_r', 'GEOSHasZ_r', 'GEOSRelatePattern_r', 'GEOSSTRtree_remove_r', 'GEOSOffsetCurve_r', 'GEOSisValid_r', 'GEOSGetSRID_r', 'GEOSWKTReader_create_r', 'GEOSWKBWriter_writeHEX_r', 'GEOSGeomToWKT_r', 'GEOSTouches_r', 'GEOSIntersection_r', 'GEOSGeom_getDimensions_r', 'GEOSPreparedDisjoint_r', 'GEOSCoveredBy_r', 'GEOSGeom_getCoordSeq_r', 'GEOSGeomTypeId_r', 'GEOSSetSRID_r', 'GEOSRelate_r', 'GEOSEquals_r', 'GEOSCoordSeq_create_r', 'GEOSSTRtree_nearest_generic_r', 'GEOSSimplify_r', 'GEOSWKBWriter_create_r', 'GEOSWKBWriter_setOutputDimension_r', 'GEOSWKTWriter_setOld3D_r', 'GEOSGeomFromWKT_r', 'GEOSGeom_createEmptyCollection_r', 'GEOSWithin_r', 'GEOSPreparedOverlaps_r', 'GEOSBuffer_r', 'GEOSClipByRect_r', 'GEOSWKBWriter_getOutputDimension_r', 'GEOSBufferWithParams_r', 'GEOSBufferWithStyle_r', 'GEOSBufferParams_setEndCapStyle_r', 'GEOSCoordSeq_setOrdinate_r', 'GEOSGetGeometryN_r', 'GEOSSTRtree_destroy_r', 'GEOSSharedPaths_r', 'GEOSBufferParams_setQuadrantSegments_r', 'GEOSWKBWriter_setByteOrder_r', 'GEOSInterpolate_r', 'GEOSMinimumClearance_r', 'GEOSWKBWriter_destroy_r', 'GEOSWKTWriter_getOutputDimension_r', 'GEOSGeom_createPolygon_r', 'GEOSProject_r', 'GEOSPreparedCrosses_r', 'GEOSGetInteriorRingN_r', 'initGEOS_r', 'GEOSHausdorffDistance_r', 'GEOSWKTReader_read_r', 'GEOSUnaryUnion_r', 'GEOSNearestPoints_r', 'GEOSWKBReader_destroy_r', 'GEOSUnion_r', 'GEOSWKTWriter_setRoundingPrecision_r', 'GEOSDisjoint_r', 'GEOSisSimple_r', 'GEOSEnvelope_r', 'GEOSisValidReason_r', 'GEOSPrepare_r', 'GEOSGetNumInteriorRings_r', 'GEOSFree_r', 'GEOSSTRtree_insert_r', 'GEOSWKBReader_create_r', 'GEOSPreparedWithin_r', 'GEOSWKTWriter_setTrim_r', 'GEOSBoundary_r', 'GEOSWKBWriter_write_r', 'GEOSArea_r', 'GEOSSTRtree_query_r', 'GEOSPointOnSurface_r', 'GEOSGetCentroid_r', 'GEOSEqualsExact_r', 'GEOSWKTWriter_create_r', 'GEOSSTRtree_create_r', 'GEOSDelaunayTriangulation_r', 'GEOSGeom_createLinearRing_r', 'GEOSCoordSeq_getDimensions_r', 'GEOSGetExteriorRing_r', 'GEOSCoordSeq_getSize_r', 'GEOSWKTWriter_write_r', 'GEOSPolygonize_r', 'GEOSLineMerge_r', 'GEOSVoronoiDiagram_r', 'GEOSWKTWriter_setOutputDimension_r', 'GEOSWKBWriter_getByteOrder_r', 'GEOSCoordSeq_setX_r', 'GEOSPreparedTouches_r', 'GEOSTopologyPreserveSimplify_r', 'GEOSCoordSeq_destroy_r', 'GEOSCoordSeq_getX_r', 'GEOSInterpolateNormalized_r', 'GEOSWKTReader_destroy_r', 'GEOSSymDifference_r', 'GEOSCoordSeq_setZ_r', 'GEOSDifference_r', 'GEOSGeomToWKB_buf_r', 'GEOSGeom_destroy_r', 'GEOSBufferParams_create_r', 'GEOSCrosses_r', 'GEOSisClosed_r', 'GEOSPreparedGeom_destroy_r', 'GEOSBufferParams_setJoinStyle_r', 'GEOSCoordSeq_setY_r', 'GEOSBufferParams_setSingleSided_r', 'GEOSGetNumCoordinates_r', 'GEOSWKBWriter_setIncludeSRID_r', 'GEOSGeomFromWKB_buf_r', 'GEOSisRing_r', 'finishGEOS_r', 'GEOSLength_r', 'GEOSGeom_createLineString_r', 'GEOSPreparedIntersects_r', 'GEOSOverlaps_r', 'GEOSWKBReader_read_r', 'GEOSWKBReader_readHEX_r', 'GEOSCovers_r']
</details>